artemis-server/src/main/java/org/apache/activemq/artemis/core/management/impl/QueueControlImpl.java

```diff
@@ -711,6 +712,32 @@ public long countMessages(final String filterStr) throws Exception {
       }
    }
 
+   @Override
+   public String countMessagesProperty(final String filter) throws Exception {
```

Comment:
Sould parse this to a filter and then later do message match filter check.

Reply:
Look at FilterImpl.createFilter to create the filter.

Reply:
In fact it will not filter the messages, but will count the messages for your property.
In a project I have several properties that I use in selectors, this count will show me how many messages there are per property variation.

Reply:
Yes but using a filter to do this allows you to do that and more powerful, and means people can use the same filtering logic theyre used to and is well tested

Reply:
There are already counters per filter countMessages (), in this case it would be more a "group by", without filter, always counting the messages of the property.

Reply:
Thr point of filter isnt to be the filter currently on thr queue but to query / filter for this count.